src/index.js

```diff
@@ -53,7 +53,8 @@ class Dropzone extends React.Component {
     e.preventDefault();
 
     // Count the dropzone and any children that are entered.
-    ++this.enterCounter;
+    this.dragTargets = this.dragTargets.filter(el => el !== e.target);
+    this.dragTargets.push(e.target);
```

Comment:
I'm confused by this part TBH. You're filtering something and then you push to the array and mutate it. Can this be a single operation instead? `reduce` for example?